my-app/src/components/IntroductionForm.js

```diff
@@ -1,42 +1,73 @@
 import React from 'react';
 import Input from './Input';
-import PasswordValidation from './PasswordValidation';
-import EmailInput from './EmailInput';
 
 export default class IntroductionForm extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = { validationStatus: {} };
-    }
+    state = {
+        username: '',
+        email: '',
+        password: '',
+        passwordConf: ''
+    };
 
-    checkFormValidation = () => {
-        const statusObject = this.state.validationStatus;
+    checkEmailValidation = () => {
+        const emailRegExValidation = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;
+        return emailRegExValidation.test(this.state.email);
+    };
 
-        if (Object.keys(statusObject).length !== 3) return false;
+    checkPasswordValidation = () => {
+        return this.state.password === this.state.passwordConf;
+    };
 
-        for (const property in statusObject) {
-            if (statusObject[property] === false) return false;
-        }
-        return true;
+    checkFormValidation = () => {
+        return this.state.username && this.checkEmailValidation() && this.checkPasswordValidation();
     };
 
-    handleValidationStatusChange = (inputStatus) => {
-        this.setState({ validationStatus: { ...this.state.validationStatus, ...inputStatus } });
+    handleInputValueChange = (key, value) => {
+        this.setState({ [key]: value });
     };
 
     handleSubmit = (event) => {
         event.preventDefault();
         if (this.checkFormValidation()) {
-            this.props.handleFormChange();
+            this.props.onChange('company');
+        }
+    };
+
+    checkPasswordStrength = (password) => {
```

Comment:
ja bym wogóle te regexy powynosił do osobnych plików np. do folderu `/utils` i napisałbym do nich testy unitowe.
poza tym regexy to jedyne miejsce, gdzie warto doddać komentarz co one robią właściwie